examples/stable-diffusion/text_to_image_generation.py

```diff
@@ -320,6 +325,12 @@ def main():
     if args.throughput_warmup_steps is not None:
         kwargs_call["throughput_warmup_steps"] = args.throughput_warmup_steps
 
+    if args.deterministic:
```

Comment:
This argument usage does not make a good case.   I would strongly recommend to rename the argument to something like `--use_cpu_rng` for allowing user to switch between HPU-based and CPU-based random number generator.   Determinism is and should be controlled only by seed, not by hard-coding seeds in a loop.   If user needs deterministic behavior he or she should control that by seed.  This sample can be deterministic if same seed is used regardless of if its HPU or CPU random number generator.

Reply:
The issue here is that setting the generator on HPU doesn't ensure determinism right?
Does it slow the whole process down to set the generator on CPU?

Reply:
> This argument usage does not make a good case. I would strongly recommend to rename the argument to something like `--use_cpu_rng` for allowing user to switch between HPU-based and CPU-based random number generator. Determinism is and should be controlled only by seed, not by hard-coding seeds in a loop. If user needs deterministic behavior he or she should control that by seed. This sample can be deterministic if same seed is used regardless of if its HPU or CPU random number generator.

We cannot use HPU-based generator. It will have the following error:
```
RuntimeError: Device type HPU is not supported for torch.Generator() api.
```

It is  intended to be an example code for users. Users can pass in any seed/value, as long as they use CPU generator. The for loop here is just a demonstration. It is expected that users will put different values. 

If we use the same value and put  (i.e. --use_cpu_rng) into the generator list, all the outputs are the same. This contradicts the intention of ```args.num_images_per_prompt```. Therefore I added a simple heuristic (the "hard-coded" for loop) to maintain determinism and diversity.

Reply:
> The issue here is that setting the generator on HPU doesn't ensure determinism right? Does it slow the whole process down to set the generator on CPU?

Similar. HPU does not implement the torch.Generator() API. Therefore we have to use CPU.

It will not affect the performance/throughput.

Reply:
> > This argument usage does not make a good case. I would strongly recommend to rename the argument to something like `--use_cpu_rng` for allowing user to switch between HPU-based and CPU-based random number generator. Determinism is and should be controlled only by seed, not by hard-coding seeds in a loop. If user needs deterministic behavior he or she should control that by seed. This sample can be deterministic if same seed is used regardless of if its HPU or CPU random number generator.
> 
> We cannot use HPU-based generator. It will have the following error:
> 
> ```
> RuntimeError: Device type HPU is not supported for torch.Generator() api.
> ```
> 
> It is intended to be an example code for users. Users can pass in any seed/value, as long as they use CPU generator. The for loop here is just a demonstration. It is expected that users will put different values.
> 
> If we use the same value and put (i.e. --use_cpu_rng) into the generator list, all the outputs are the same. This contradicts the intention of `args.num_images_per_prompt`. Therefore I added a simple heuristic (the "hard-coded" for loop) to maintain determinism and diversity.

Using generator via `set_seed()` from `optimum.habana.utils` should be a default.  You are adding a CPU based RNG generator for which you also can set seed.   Determinism is controlled by setting the same seed AND using same RNG.  So why not have `--seed` and `--use_cpu_rng` for these selections?

When user sets the same seed once, the RNG should spit out a deterministic sequence of pseudorandom numbers, so no need to set seed in a loop for each sample individually.

Reply:
OK. I will change the wordings from `--deterministic` to `--use_cpu_rng`. Also the CPU generator will take the values from `--seed`

Reply:
thanks @cfgfung  but now the value of `manual_seed(args.seed)` is independent of `i`. 
Should we change that to `manual_seed(args.seed+i)`?

Reply:
@imangohari1 
Yes, it will be much better. It will be able to generate different images. I have added that.